jsonschema/tests/test_cli.py

```diff
@@ -149,3 +154,20 @@ def test_version(self):
         )
         version = version.decode("utf-8").strip()
         self.assertEqual(version, __version__)
+
+    def test_piping(self):
+        sys.stdin = StringIO("{}")
```

Comment:
Thanks, this definitely now doesn't use mock, but it does now leak global state across test runs -- from here on out, every test will see `sys.stdin` as being the StringIO here.

What I had in mind was more that you should pass the input into the `cli.run` call below.

Specifically, if we add a `stdin` argument to `run` (with a default of `sys.stdin`), then you can pass your `NativeIO` for `stdin` to `run`, same as how `stdout` and `stderr` are being passed to it.